x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -198,3 +209,24 @@ export const getEndpointAuthzInitialState = (): EndpointAuthz => {
     canWriteEndpointExceptions: false,
   };
 };
+
+/**
+ * Duplicate logic to calculate if user has privilege to fetch Agent Policies,
+ * working only with Capabilities, in order to be able to use it e.g. in middleware.
+ *
+ * The logic works with Fleet granular privileges (`subfeaturePrivileges`) both enabled and disabled.
+ *
+ * @param capabilities Capabilities from coreStart.application
+ */
+export const canFetchPackageAndAgentPolicies = (capabilities: Capabilities): boolean => {
+  const canReadPolicyManagement = (capabilities.siem?.readPolicyManagement ?? false) as boolean;
+
+  const fleetv2 = capabilities.fleetv2;
+  const canReadFleetAgentPolicies = (fleetv2?.read &&
+    (fleetv2?.agent_policies_read === true ||
```

Comment:
nit: A little bit hard to follow these conditions, wouldn't it come down to 
`(capabilities.fleetv2?.agent_policies_read ?? true)` ?

Reply:
yes, it comes to that. the idea was to be able to read it as we accept either `true` or `undefined`, but yeah, maybe simply a fallback to the `true` value is more readable
de27c61af72d2a7e11ad045c31dc8239300cebab